app/controllers/social_previews_controller.rb

```diff
@@ -1,28 +1,65 @@
 class SocialPreviewsController < ApplicationController
   # No authorization required for entirely public controller
 
+  PNG_CSS = "body { transform: scale(0.3); } .preview-div-wrapper { overflow: unset; margin: 5vw; }".freeze
```

Comment:
Filling the entire viewport results in a ~4000x4000 pixel image. Scaling it down makes rendering way faster.